cinema/views.py

```diff
@@ -21,24 +25,49 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    GenericViewSet
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
add TokenAuthentication to default authentication_classes in settings.py
add IsAdminOrIfAuthenticatedReadOnly to default permission_classes  in settings.py
to not duplicate code
remove redundant 
```
    authentication_classes = (TokenAuthentication,)
    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```
from everywhere